OpenEdXMobile/src/main/java/org/edx/mobile/util/LocaleUtils.java

```diff
@@ -16,10 +16,24 @@ public static String getCountryNameFromCode(@NonNull String countryCode) throws
 
     public static String getLanguageNameFromCode(@NonNull String languageCode) throws InvalidLocaleException {
         final String lowercaseLanguage = languageCode.toLowerCase(Locale.ROOT);
-        if (!isValidBcp47Alpha(lowercaseLanguage, 2, 3)) {
-            throw new InvalidLocaleException("Invalid language: " + languageCode);
+
+        switch (lowercaseLanguage) {
+            // Chinese languages are special cases. The server uses different codes when compared
+            // to Android's locale library. Addtionally, zh_CN and zh_TW are not languageCodes that
+            // Android's locale recognizes as exceptions. Android locale only returns "Chinese" so
+            // these Chinese name variations are hardcoded in.
+            case "zh_cn":
+            case "zh_hans":
+                return "Simplified Chinese";
```

Comment:
The other option is just to use Android Locale's Locale.SIMPLIFIED_CHINESE even though the display name is just "Chinese". It would probably be more localization friendly this way.